packages/core/types/src/types/__tests__/utils/object.test.ts

```diff
@@ -77,4 +77,14 @@ describe('Utils.Object', () => {
     type('ValuesNever').isNever();
     type('ValuesContainNever').isUnion([t.stringLiteral('foo'), t.stringLiteral('bar')]);
   });
+
+  test('Replace', () => {
+    const expectedResultType = t.object({
+      properties: {
+        foo: t.numberLiteral(2),
+        bar: t.stringLiteral('foo'),
+      },
+    });
+    type('Replace').isObject(expectedResultType);
```

Comment:
@Convly This is not working at the moment, could you give me a hand on how I should implement this test? I am unsure how to test object types.
I am getting this error at the moment:
![image](https://github.com/strapi/strapi/assets/20578351/cfca61a0-c10e-4022-a530-fe2e46059642)

Reply:
Try using `t.annonymousObject` instead of `t.object`

Reply:
replacing it with anonymous object I get this error
![image](https://github.com/strapi/strapi/assets/20578351/b9f6eb15-ed1b-43b0-994b-606e15fefdd8)

Reply:
ah... same issue as for relations' tests.
Add this test as a todo test, I'll spend this Thursday & Friday fixing issues like this one & adding documentation to ts-zen.

Reply:
(Also, I believe the return type will be an intersection of a computed object & an anonymous object rather than a pure anonymous object, but I'll have to dig deeper on that kind of stuff)

Reply:
Maybe if we used the resolved type utility the return type would be a pure anonymous object:

```
/**
 * Makes types more readable if using complex
 *
 * @example
 * type X = Resolve<Replace<{ foo: 'bar', bar: 'foo' }, {foo: 2}>>
 * //  <{foo: 2, bar: 'foo'}>
 *
 * Without resolve:
 * type X = Replace<{ foo: 'bar', bar: 'foo' }, {foo: 2}>
 * //  Omit<{foo: 'bar', bar: 'foo'}, 'foo'> & { foo: 2}
 */
export type Resolve<T> = T extends (...args: unknown[]) => unknown ? T : { [K in keyof T]: T[K] };
```

Reply:
just discovered the ts-zen package was yours 🚀